### PR TITLE
feat: add vertical tabs

### PR DESCRIPTION
**Pull request by v8tenko**

### CODE REVIEW

File: src/plugin/transform.ts [hunk 1]
```diff
@@ -290,13 +331,17 @@ export function transform({
                 }
 
                 const tabsGroup = match[2] || `${DEFAULT_TABS_GROUP_PREFIX}${generateID()}`;
+                const orientation = (match[4] || 'horizontal') as TabsOrientation;
 
                 const {tabs, index} = findTabs(state.tokens, i + 3);
 
+                debugger;
```
Comment (vsesh) [inline src/plugin/transform.ts:338]: I received a notification about PR. Is it ready to review?

Comment (v8tenko) [reply src/plugin/transform.ts:338]: No, I just pushed it to check by myself, sry for notification

Comment (vsesh) [reply src/plugin/transform.ts:338]: Okay. You can call me when it will be ready to review. I'll try to make it fast

File: test/plugin.test.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ describe('plugin', () => {
             const attrsObject = convertAttrsToObject(tab);
 
             expect(Object.keys(attrsObject)).toEqual(attrs);
-            expect(attrsObject['class']).toEqual(`yfm-tab${i === 0 ? ' active' : ''}`);
+            expect(attrsObject['class']).toEqual(`yfm-tab yfm-tab-group${i === 0 ? ' active' : ''}`);
             expect(attrsObject['role']).toEqual('tab');
             expect(attrsObject['tabindex']).toEqual(i === 0 ? '0' : '-1');
         });
```
Comment (vsesh) [inline test/plugin.test.ts:78]: Please add new tests for both types of orders

File: src/runtime/TabsController.ts [hunk 3]
```diff
@@ -219,10 +280,31 @@ export class TabsController {
...
                 ? element.closest(Selector.TAB_LIST)
                 : null;
+
         return tabList?.closest(Selector.TABS);
     }
 
+    private areTabsVertical(target: HTMLElement) {
+        const parent = target.parentElement;
+
+        return target.dataset.diplodocVerticalTab || Boolean(parent?.dataset.diplodocVerticalTab);
```
Comment (vsesh) [inline src/runtime/TabsController.ts:290]: You should recheck this behavior very carefully https://github.com/diplodoc-platform/tabs-extension/blob/master/src/runtime/TabsController.ts#L201-L204

Comment (vsesh) [reply src/runtime/TabsController.ts:290]: add this too https://github.com/diplodoc-platform/tabs-extension/blob/master/src/runtime/TabsController.ts#L56

File: src/react/useDiplodocTabs.ts [hunk 4]
```diff
@@ -22,6 +22,10 @@ export function useDiplodocTabs(callback: UseDiplodocTabsCallback) {
                 window[GLOBAL_SYMBOL].selectTabById(tabId, options),
             [],
         ),
-        selectTab: useCallback((tab: Tab) => window[GLOBAL_SYMBOL].selectTab(tab), []),
+        // @todo remove
```
Comment (vsesh) [inline src/react/useDiplodocTabs.ts:25]: ? :)

File: src/plugin/transform.ts [hunk 5]
```diff
@@ -25,7 +28,9 @@ export type PluginOptions = {
     bundle: boolean;
 };
 
-const TAB_RE = /`?{% list tabs( group=([^ ]*))? %}`?/;
+export type TabsOrientation = 'vertical' | 'horizontal';
+
+const TAB_RE = /`?{% list tabs( group=([^ ]*))?( (vertical)|(horizontal))? %}`?/;
```
Comment (vsesh) [inline src/plugin/transform.ts:33]: Please add examples of new attributes in `README.md`

File: src/runtime/TabsController.ts [hunk 6]
```diff
@@ -5,12 +5,14 @@ import {
...
     TABS_CLASSNAME,
     TABS_LIST_CLASSNAME,
+    TABS_VERTICAL_CLASSNAME,
     TAB_CLASSNAME,
     TAB_DATA_ID,
     TAB_DATA_KEY,
     TAB_PANEL_CLASSNAME,
     Tab,
 } from '../common';
+import {TabsOrientation} from '../plugin/transform';
```
Comment (vsesh) [inline src/runtime/TabsController.ts:15]: import type

**Review by vsesh — APPROVED** [review]

**Pull request merged**